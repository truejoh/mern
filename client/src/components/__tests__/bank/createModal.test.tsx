import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import EditModal from '../../bank/modals/editModal';

describe('Bank Create Modal', () => {
  it('renders component', () => {
    const props = {
      visible: true,
      item: {
        _id: '123',
        name: 'Bank A',
        branchName: 'Branch A',
      },
      onOk: () => {},
      onCancel: () => {},
    };
    const component = shallow(<EditModal {...props} />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders props', () => {
    const props = {
      visible: true,
      item: {
        _id: '123',
        name: 'Bank A',
        branchName: 'Branch A',
      },
      onOk: () => {},
      onCancel: () => {},
    };
    const component = shallow(<EditModal {...props} />);

    expect(component.prop('visible')).toEqual(true);
    expect(component.prop('onOk')).toBeInstanceOf(Function);
    expect(component.prop('onCancel')).toBeInstanceOf(Function);
  });
});
