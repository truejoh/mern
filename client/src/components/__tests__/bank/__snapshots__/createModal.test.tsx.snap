// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bank Create Modal renders component 1`] = `
<Modal
  confirmLoading={false}
  maskTransitionName="fade"
  okText="Save"
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  title={
    <InjectIntl(InjectMassage)
      id="bank.modal.edit.title"
    />
  }
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Form
    colon={true}
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
  >
    <FormItem
      hasFeedback={false}
      label={
        <InjectIntl(InjectMassage)
          id="bank.modal.name"
        />
      }
    >
      <Input
        defaultValue="Bank A"
        disabled={false}
        onChange={[Function]}
        placeholder="input bank name"
        type="text"
      />
    </FormItem>
    <FormItem
      hasFeedback={false}
      label={
        <InjectIntl(InjectMassage)
          id="bank.modal.branch"
        />
      }
    >
      <Input
        defaultValue="Branch A"
        disabled={false}
        onChange={[Function]}
        placeholder="input branch"
        type="text"
      />
    </FormItem>
  </Form>
</Modal>
`;
